File: backend/src/models/Service.js
const { Model, DataTypes } = require('sequelize');

class Service extends Model {
  static initModel(sequelize) {
    Service.init({

  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  name: {
    type: DataTypes.STRING,
    allowNull: false,
    unique: true
  },
  description: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  category: {
    type: DataTypes.STRING,
    allowNull: false
  },
  price: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false,
    defaultValue: 0
  },
  duration: {
    type: DataTypes.INTEGER, // in minutes
    allowNull: true
  },
  isActive: {
    type: DataTypes.BOOLEAN,
    defaultValue: true
  },
  image: {
    type: DataTypes.TEXT, // URL or base64
    allowNull: true
  },
  requirements: {
    type: DataTypes.JSON, // Array of requirements
    allowNull: true
  },
  specializations: {
    type: DataTypes.JSON, // Array of required specializations
    allowNull: true
  },
  // Additional fields for better doctor matching
  minExperience: {
    type: DataTypes.INTEGER,
    allowNull: true
  },
  maxPrice: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: true
  },
  location: {
    type: DataTypes.STRING,
    allowNull: true
  }

    }, {
      sequelize,
      modelName: 'Service',

  timestamps: true,
  tableName: 'services'

    });
    return Service;
  }

  static associate(models) {
    // Associations will be defined in models/index.js
  }
}

module.exports = Service;

File: fastapi_backend/models/service.py
from beanie import Document
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime
from decimal import Decimal

# Pydantic schemas
class ServiceBase(BaseModel):
    name: str
    description: Optional[str] = None
    category: str
    price: Decimal = Field(default=Decimal("0.00"), ge=0)
    duration: Optional[int] = None  # in minutes
    is_active: bool = True
    image: Optional[str] = None
    requirements: Optional[List[str]] = None
    specializations: Optional[List[str]] = None
    min_experience: Optional[int] = None
    max_price: Optional[Decimal] = None
    location: Optional[str] = None

class ServiceCreate(ServiceBase):
    pass

class ServiceUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    category: Optional[str] = None
    price: Optional[Decimal] = Field(None, ge=0)
    duration: Optional[int] = None
    is_active: Optional[bool] = None
    image: Optional[str] = None
    requirements: Optional[List[str]] = None
    specializations: Optional[List[str]] = None
    min_experience: Optional[int] = None
    max_price: Optional[Decimal] = None
    location: Optional[str] = None

class ServiceResponse(ServiceBase):
    id: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

# Beanie Document
class Service(Document, ServiceBase):
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

    class Settings:
        name = "services"
        indexes = [
            "name",
            "category",
            "is_active",
            "specializations",
            "location",
            "created_at"
        ]

    def __str__(self):
        return f"Service(id={self.id}, name={self.name}, category={self.category})"
